src/ecdsa/keys.py

```diff
@@ -1133,3 +1133,22 @@ def sign_number(self, number, entropy=None, k=None):
         assert 1 <= _k < order
         sig = self.privkey.sign(number, _k)
         return sig.r, sig.s
+
+    def ecdh_get_shared_secret(self, remote_public_key):
+        """
+        Elliptic-curve Diffie–Hellman (ECDH). A key agreement protocol.
+        Allows two parties, each having an elliptic-curve public–private key 
+        pair, to establish a shared secret over an insecure channel
+                
+        :param remote_public_key: public key of remote party (VerifyingKey)
+        :return: shared secret
+        """""
+        # shared secret = PUBKEYtheirs * PRIVATEKEYours
+        result = remote_public_key.pubkey.point * self.privkey.secret_multiplier
```

Comment:
result needs to be checked if it's not INFINITY and raise exception if it is

see 2nd action in [Section 3.3.1 of SEC1 v2](http://www.secg.org/sec1-v2.pdf)